### PR TITLE
Log run data

### PR DESCRIPTION
**Pull request by mociarain**

### CODE REVIEW

File: mlflow_faculty/trackingstore.py [hunk 1]
```diff
@@ -292,5 +273,38 @@ def search_runs(self, experiment_ids, search_expressions, run_view_type):
...
 
-    def log_batch(self, run_id, metrics, params, tags):
-        raise NotImplementedError()
+    def log_batch(self, run_uuid, metrics=None, params=None, tags=None):
+        """
+        Fetches the experiment by ID from the backend store.
+
+        :param experiment_id: Integer id for the experiment
+        :param run_uuid: string containing run UUID
+            (32 hex characters = a uuid4 stripped off of dashes
```
Comment (acroz) [inline mlflow_faculty/trackingstore.py]: Missing `)`?

File: mlflow_faculty/trackingstore.py [hunk 2]
```diff
@@ -292,5 +273,38 @@ def search_runs(self, experiment_ids, search_expressions, run_view_type):
         else:
             return [faculty_run_to_mlflow_run(run) for run in faculty_runs]
 
-    def log_batch(self, run_id, metrics, params, tags):
-        raise NotImplementedError()
+    def log_batch(self, run_uuid, metrics=None, params=None, tags=None):
+        """
+        Fetches the experiment by ID from the backend store.
+
+        :param experiment_id: Integer id for the experiment
```
Comment (acroz) [inline mlflow_faculty/trackingstore.py]: Remove this line?

File: mlflow_faculty/trackingstore.py [hunk 3]
```diff
@@ -292,5 +273,38 @@ def search_runs(self, experiment_ids, search_expressions, run_view_type):
...
+        :param run_uuid: string containing run UUID
+            (32 hex characters = a uuid4 stripped off of dashes
+        :param metrics: List of Mlflow Metric entities.
+        :param params: List of Mlflow Param entities
+        :param tags: List of Mlflow Tag entities.
+        """
+        metrics = set_to_empty_list_none(metrics)
```
Comment (acroz) [inline mlflow_faculty/trackingstore.py]:
```suggestion
        metrics = [] if metrics is None else list(metrics)
```

File: mlflow_faculty/trackingstore.py (hunk 2 above)
Comment (mociarain) [reply mlflow_faculty/trackingstore.py]: Between the description and the param list?

File: mlflow_faculty/trackingstore.py (hunk 1 above)
Comment (mociarain) [reply mlflow_faculty/trackingstore.py]: 👍

File: mlflow_faculty/trackingstore.py (hunk 3 above)
Comment (mociarain) [reply mlflow_faculty/trackingstore.py]: That's better than I thought it would look. Will do

File: tests/test_trackingstore.py [hunk 4]
```diff
@@ -435,3 +437,94 @@ def test_search_runs_client_error(mocker):
             search_expressions=None,
             run_view_type=None,
         )
+
+
+def test_log_batch(mocker):
+    mock_client = mocker.Mock()
+    mocker.patch("faculty.client", return_value=mock_client)
+    mock_mlflow_metric = mocker.Mock()
```
Comment (acroz) [inline tests/test_trackingstore.py]: This mock isn't really being used (also the other 2 below)

File: tests/test_trackingstore.py [hunk 5]
```diff
@@ -435,3 +437,94 @@ def test_search_runs_client_error(mocker):
...
+    mocker.patch("faculty.client", return_value=mock_client)
+
+    store = FacultyRestStore(STORE_URI)
+
+    with pytest.raises(MlflowException, match="param-key"):
+        with pytest.raises(
+            faculty.clients.experiment.ParamConflict, match="message"
+        ):
+            store.log_batch(EXPERIMENT_RUN_UUID_HEX_STR)
+    mock_client.log_run_data.assert_called_once_with(
```
Comment (acroz) [inline tests/test_trackingstore.py]: Could omit this assertion if you want

File: tests/test_trackingstore.py [hunk 6]
```diff
@@ -435,3 +437,94 @@ def test_search_runs_client_error(mocker):
...
+        message="message", conflicting_params=["param-key"]
+    )
+
+    mock_client.log_run_data.side_effect = exception
+    mocker.patch("faculty.client", return_value=mock_client)
+
+    store = FacultyRestStore(STORE_URI)
+
+    with pytest.raises(MlflowException, match="param-key"):
+        with pytest.raises(
```
Comment (acroz) [inline tests/test_trackingstore.py]: Not sure we want this inner `pytest.raises`

File: tests/test_trackingstore.py [hunk 7]
```diff
@@ -435,3 +437,94 @@ def test_search_runs_client_error(mocker):
...
+        mocker.Mock(), error="error_message", error_code="some_error_code"
+    )
+
+    mock_client.log_run_data.side_effect = exception
+    mocker.patch("faculty.client", return_value=mock_client)
+
+    store = FacultyRestStore(STORE_URI)
+
+    with pytest.raises(MlflowException, match="error_message"):
+        with pytest.raises(HttpError, match="some_error_code"):
```
Comment (acroz) [inline tests/test_trackingstore.py]: Ditto (on the inner `pytest.raises`

File: mlflow_faculty/trackingstore.py (hunk 3 above)
Comment (mociarain) [reply mlflow_faculty/trackingstore.py]: I implemented this was unable to remove the list enclosing `map()` sine in python3 this returns a map object

**Pull request merged**